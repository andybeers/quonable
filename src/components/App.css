.flexWrapper {
  display: flex;
  flex-direction: column;
}

main {
  min-height: 75vh;
  flex: 1;
}