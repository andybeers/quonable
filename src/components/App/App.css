.flexWrapper {
  display: flex;
  flex-direction: column;
}

main {
  min-height: 95vh;
  flex: 1;
}