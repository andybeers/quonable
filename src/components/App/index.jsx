import React, { Component } from 'react';
import './App.css';
import Quote from '../Quote';
import AppFooter from '../AppFooter';
import authors from '../../authors';
import quotes from '../../quotes';

class App extends Component {
  state = {
    showSeriousQuote: false,
    quoteReal: { quote: 'Init', author: 'Init' },
    quoteGoofy: { quote: 'Init', author: 'Init' },
  }

  newQuote = () => {
    let quoteRealRand = this.roll(quotes.real.length);
    let quoteGoofyRand = this.roll(quotes.goofy.length);
    let authRealRand = this.roll(authors.real.length);
    let authGoofyRand = this.roll(authors.goofy.length);
    let quoteReal = {
      quote: quotes.real[quoteRealRand].text,
      author: authors.goofy[authGoofyRand].name
    };
    let quoteGoofy = {
      quote: quotes.goofy[quoteGoofyRand].text,
      author: authors.real[authRealRand].name
    };

    this.setState({
      quoteReal: quoteReal,
      quoteGoofy: quoteGoofy,
    });
  }

  changeView = () => {
    this.setState(prevState => ({ showSeriousQuote: !prevState.showSeriousQuote }));
  }

  roll(maxRange) {
    return Math.floor(Math.random() * maxRange);
  }

  componentDidMount() {
    this.newQuote();
  }

  render() {
    return (
        <div className='flexWrapper'>
          <Quote 
            showSeriousQuote={this.state.showSeriousQuote}
            quoteReal={this.state.quoteReal}
            quoteGoofy={this.state.quoteGoofy}
            newQuote={this.newQuote}
            changeView={this.changeView}
          />
          <AppFooter />
        </div>
    );
  }
}

export default App;
