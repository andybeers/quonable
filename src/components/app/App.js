import React, { Component } from 'react';
import './App.css';
import Quote from '../quote/Quote';

class App extends Component {
  render() {
    return (
      <div>
        <Quote />
      </div>
    );
  }
}

export default App;
