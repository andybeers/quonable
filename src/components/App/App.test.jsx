import React from "react";
import ReactDOM from "react-dom";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

import App from "./App";

describe("Top level App", () => {
  it("Renders App correctly", () => {
    const rendered = shallow(<App />);
    expect(toJson(rendered)).toMatchSnapshot();
  });
});
