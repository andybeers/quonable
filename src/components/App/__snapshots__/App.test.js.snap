exports[`Top level App Renders App correctly 1`] = `
<div
  className="flexWrapper">
  <Quote
    changeView={[Function]}
    newQuote={[Function]}
    quoteGoofy={
      Object {
        "author": "Init",
        "quote": "Init",
      }
    }
    quoteReal={
      Object {
        "author": "Init",
        "quote": "Init",
      }
    }
    showSeriousQuote={false} />
  <AppFooter />
</div>
`;
